.travis/linux/OpenBLAS/install.sh

```diff
@@ -1,31 +1,58 @@
 set -ex
 
+CACHE_DIR=${TRAVIS_BUILD_DIR}/.travis/${BLAS_LIB}.cache
+
 # fetch fortran to build OpenBLAS
-sudo apt-get update -qq
-sudo apt-get install -qq gfortran
-
-# fetch OpenBLAS
-pushd ~
-sudo git clone --depth=1 git://github.com/xianyi/OpenBLAS
-
-# make OpenBLAS
-pushd OpenBLAS
-echo OpenBLAS $(git rev-parse HEAD)
-sudo make FC=gfortran &> /dev/null
-sudo make PREFIX=/usr install
-popd
+sudo apt-get update -qq && sudo apt-get install -qq gfortran
 
-# fetch cblas reference lib
-curl http://www.netlib.org/blas/blast-forum/cblas.tgz | tar -zx
+# check if cache exists
+if [ -e ${CACHE_DIR}/last_commit_id ]; then
+    echo "Cache $CACHE_DIR hit"
+    LAST_COMMIT="$(git ls-remote git://github.com/xianyi/OpenBLAS HEAD | grep -o '^\S*')"
+    CACHED_COMMIT="$(cat ${CACHE_DIR}/last_commit_id)"
+    # determine current OpenBLAS master commit id and compare
+    # with commit id in cache directory
+    if [ "$LAST_COMMIT" != "$CACHED_COMMIT" ]; then
+        echo "Cache Directory $CACHE_DIR has stale commit"
+        # if commit is different, delete the cache
+        rm -rf ${CACHE_DIR}
+    fi
+fi
 
-# make and install cblas
-pushd CBLAS
-sudo mv Makefile.LINUX Makefile.in
-sudo BLLIB=/usr/lib/libopenblas.a make alllib
-sudo mv lib/cblas_LINUX.a /usr/lib/libcblas.a
-popd
-popd
+# check if cache directory exists
+if [ ! -e ${CACHE_DIR}/last_commit_id ]; then
+    if [ -d ${CACHE_DIR} ]; then
+        # Travis automatically creates the cache directory if it does not exist,
+        # so this is needed for initialization.
+        rm -rf ${CACHE_DIR}
+    fi
+
+    # cache generation
+    echo "Building cache at $CACHE_DIR"
+    mkdir ${CACHE_DIR}
+    sudo git clone --depth=1 git://github.com/xianyi/OpenBLAS
+
+    pushd OpenBLAS
+    echo OpenBLAS $(git rev-parse HEAD)
+    # write commit id to cache location
+    echo $(git rev-parse HEAD) > ${CACHE_DIR}/last_commit_id
+    sudo make FC=gfortran &> /dev/null && sudo make PREFIX=${CACHE_DIR} install
```

Comment:
Sorry, now I understand. For some reason I thought that OpenBLAS and CBLAS are subdirectories of CACHE_DIR.